softwarecitationauditor/openai_checker.py

```diff
@@ -9,58 +22,102 @@ def load_prompt_template():
     with open(prompt_path, "r") as f:
         return f.read()
 
-def extract_and_check_software(body_text, bibliography_text, pdf_filename, provider, model, client=None):
+console = Console()
+
+def print_json_as_table(json_data, step_number):
+    if isinstance(json_data, dict):
```

Comment:
This code can be improved. In both cases, the code is creating keys and then adding rows into a table. Refactor so that this logic is not repeated in each clause.